dips/dip-158.md

```diff
@@ -0,0 +1,794 @@
+* `InfoCommand` - Allows obtianing payment information based on reference id
+* `InitCommand` - Initializes the payment process
+* `ReadyCommand` - Notifies the payment is ready for settlement on-chain
+* `AuthorizedCommand` - Notifies that payment funds are authorized (locked) for future capture
+* `CaptureCommand` - Request to capture previously authorized (locked) funds 
+* `VoidCommand` - Voids a payment
+* `AbortCommand` - Aborts a payment
+
+Accordingly, the following list specifies additional values for `_ObjectType` field of  `CommandObject`
+* `InfoCommand` 
+* `InitCommand` 
+* `ReadyCommand`
+* `AuthorizedCommand` 
+* `CaptureCommand`
+* `VoidCommand`
+* `AbortCommand`
+
+## PaymentInfoObject (new)
+The minimal payment details required in order for the payer to decide whether the accept the payment or reject it. 
+
+Field|Type|Required?|Description|
+|-|-|-|-|
+|receiver|PaymentActorObject|Y|Only the address field of the PaymentActorObject is required. i.e. status element is not required|
+|action|PaymentActionObject|Y|Information regarding the type of payment and payment amount|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|description|str|N|Description of the payment. To be displayed to the user|
+|
+
+```
+"payment_info": 
+{
+  "receiver": {
+    "address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+    "merchant_data": { 
+      "name": "Uber",
+      "legal_name": "Uber Technologies Inc.",
+      "address": {
+        "city": "San Francisco",
+        "country": "US",
+        "line1": "1455 Market Street",
+        "line2": "Suite 400",
+        "postal_code": "94103",
+        "state": "CA"
+      }        
+    }        
+  },
+  "action": {
+    "amount": "100000000",
+    "currency": "XUS",
+    "action": "charge",
+    "timestamp": "72322"
+  },
+  "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+  "description": "payment request description"     
+}
+```
+
+## MerchantDataObject (new)
+The details of the merchant.
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|name|str|Y|Merchant's display name. Should be recognizable by the payer|
+|legal_name|str|Y|The legal entity name|
+|address|AddressObject|Y|Merchant's physical address information|
+|
+```
+"merchant_data": 
+{
+  "name": "Uber",
+  "legal_name": "Uber Technologies Inc.",
+  "address": {
+    AddressObject()
+  }        
+}
+``` 
+
+## PayerDataObject (new)
+The details of the payer. payer_data is the minimal set of payer information required by the merchant to perform risk checks and apply AML policies.  
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|given_name|str|Y|Legal given name of the payer|
+|surname|str|Y|Legal surname of the payer|
+|address|AddressObject|Y|Merchant's physical address information|
+|national_id|NationalIdObject|N|National ID information for payer. e.g. SSN|
+|
+```
+"payer_data": 
+{
+  "name": "Ben",
+  "surname": "Maurer",
```

Comment:
It might be better to use fake names

Reply:
Was not aware this is a real name (taken from DIP-1). I will change.